src/main/java/com/google/cloud/tools/gradle/appengine/core/AppEngineCorePluginConfiguration.java

```diff
@@ -79,6 +82,32 @@ public void execute(Project project) {
         });
   }
 
+  private void createManagedSdkTask() {
+    ManagedSdkTask managedSdkTask =
+        project
+            .getTasks()
+            .create(
+                MANAGED_SDK_TASK_NAME,
+                ManagedSdkTask.class,
+                new Action<ManagedSdkTask>() {
+                  @Override
+                  public void execute(final ManagedSdkTask managedSdkTask) {
+                    managedSdkTask.setGroup(taskGroup);
+                    managedSdkTask.setDescription("Download the Cloud SDK");
+
+                    project.afterEvaluate(
+                        new Action<Project>() {
+                          @Override
+                          public void execute(Project p) {
+                            managedSdkTask.setToolsConfig(toolsExtension);
+                          }
+                        });
+                  }
+                });
+
+    project.getTasks().getByName(BasePlugin.ASSEMBLE_TASK_NAME).dependsOn(managedSdkTask);
```

Comment:
I'm probably just missing something, but what is the purpose of attaching the assemble task to the managedSdkTask?

Reply:
This was a bit of a guess on my part; I wanted to make sure the cloud sdk is downloaded before the project as built, so I have it run before the "assemble" task. Is there a better place to put it in the task tree?

Reply:
Hmm, I believe there isn't a need to have assemble happen after the managedCloudSdk task (building the project artifacts shouldn't depend on there being Cloud SDK). However, the other tasks in this plugin would need Cloud SDK to be present. @loosebazooka should there be a hard dependency from the other tasks to the managedCloudSdk task?

Reply:
Yeah it might not actually be ideal. We could probably have the tasks we know that need the cloud sdk just depend on this task.

like deploy/run/stage, there's also a way to write task addition callbacks. so when any task that is added that starts with "appengine", you make it depend on managedCloudSdk.